src/pages/sitemap-docs.xml.tsx

```diff
@@ -0,0 +1,31 @@
+import { GetServerSidePropsContext } from 'next'
+
+export default function SiteMap() {
+	return null
+}
+
+const siteUrl = process.env.SITE_URL || 'https://developer.hashicorp.com'
+
+export async function getServerSideProps({ res }: GetServerSidePropsContext) {
+	const response = await fetch(
+		`https://content.hashicorp.com/api/all-docs-paths`
+	)
+	const { result } = await response.json()
+	const createSitemap = () => `<?xml version="1.0" encoding="UTF-8"?>
+    <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+        ${result
+					.map(
+						(sitePath: { path: string; created_at: string }) =>
+							`<url><loc>${siteUrl}/${sitePath.path}</loc><lastmod>${sitePath.created_at}</lastmod><changefreq>daily</changefreq><priority>1.0</priority></url>
+              `
+					)
+					.join('')}
+    </urlset>
+    `
+	res.setHeader('Content-Type', 'text/xml')
```

Comment:
It seems like we should add a cache header here 🤔 ([docs](https://vercel.com/docs/concepts/edge-network/caching#serverless-functions)).

I think ideally we'd have a fairly long TTL, like an hour?

Reply:
Removed file in favor of next-sitemap, but thanks for the comment!

Reply:
@nayres I think we still want to set a cache header here, it doesn't look like `next-sitemap` sets one: https://github.com/iamvishnusankar/next-sitemap/blob/master/packages/next-sitemap/src/ssr/response.ts